thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/DetectionTaskRunner.java

```diff
@@ -91,6 +98,10 @@ public DetectionTaskRunner() {
     Map<DimensionKey, MetricTimeSeries> dimensionKeyMetricTimeSeriesMap =
         TimeSeriesUtil.getTimeSeriesForAnomalyDetection(anomalyFunctionSpec, startEndTimeRanges);
 
+    overrideConfigs = OverrideConfigHelper.getTimeSeriesOverrideConfigs(
```

Comment:
this is inefficient, it will make one query for every time bucket and check if there is an override.

its also not clear if the configoverride is specified for every time interval or only the start time and end time.

I think a better logic is to query for all overideConfigs that overlap with the window start/end time range. If you dont find anything we should simply continue with existing logic.